### DISCUSSION TITLE
Is there any equivalent to vtkRectilinearGridGeometryFilter in PyVista?

### DISCUSSION THREAD

**Discussion by toro01** [pyvista/pyvista#3228] [Q&A] [answered]
Hi all, thanks for the great work in PyVista!

I want to migrate a filter sequence written with vtk to PyVista. One of the filter used is vtkRectilinearGridGeometryFilter. It removes all but the geometry described as an extent from a rectilinear grid. I could find the clip filter in PyVista that does almost the opposite. 

Is there any equivalent to vtkRectilinearGridGeometryFilter in PyVista?

Thanks for your help.

**Comment by banesullivan** [comment, marked as answer]
Not at this time, but we can certainly implement this! Can you point us to an example of `vtkRectilinearGridGeometryFilter`?

**Reply by toro01** [reply]
Hi, I can give a use case.

I've a rectilinear grid with at most 1 billion points. An array representing temperatures of the points is attached to this grid. I want to calculate the min/average/max temperature of arbitrary areas in this rectilinear grid (of 2d/3d subareas of the rectilinear grid). For that the area of the grid matching a position/size specified y the user, must be filtered using `vtkRectilinearGridGeometryFilter`

The algorithm is:
- get the extent in the grid of a geometry specified by the user (position and size)
- "extract" from the grid the area that matches this extent. Using VTK Python, this looks like:
```
filter = vtkRectilinearGridGeometryFilter()
filter.SetInputData(my_rect_grid)
filter.SetExtent(extent)
filter.Update()
```
- to calculate min and max, I just get the value range of the temperatures array:
```
filter.GetOutput()
.GetPointData()
.GetArray("temperatures")
.GetValueRange()
```
- to calculate average, additional steps are required, but still I need to filter the rectilinear grid.

Let my know if this is not clear or if you see an alternative way of doing this.

**Reply by banesullivan** [reply]
I think this is pretty clear - do you expect the `vtkRectilinearGridGeometryFilter` to return the points of the mesh and no actual geometry?

**Reply by banesullivan** [reply]
FYI: https://github.com/pyvista/pyvista/pull/3256

**Reply by toro01** [reply]
On the result, I need to be able to apply a point to cell filter, then get the bounds of the cell to calculate the surface of each cell. So I guess I need what you call the geometry. Does it answer your question?
